docs/src/components/showcase/FolderTreeView.tsx

```diff
@@ -2,209 +2,275 @@
 import * as React from 'react';
 import clsx from 'clsx';
+import { animated, useSpring } from '@react-spring/web';
 import { styled } from '@mui/material/styles';
+import { TransitionProps } from '@mui/material/transitions';
 import Box from '@mui/material/Box';
-import { TreeView } from '@mui/x-tree-view/TreeView';
-import {
-  TreeItem,
-  useTreeItem,
-  TreeItemProps,
-  TreeItemContentProps,
-} from '@mui/x-tree-view/TreeItem';
+import Collapse from '@mui/material/Collapse';
 import Typography from '@mui/material/Typography';
 import KeyboardArrowDownRounded from '@mui/icons-material/KeyboardArrowDownRounded';
 import KeyboardArrowUpRounded from '@mui/icons-material/KeyboardArrowUpRounded';
 import FolderRounded from '@mui/icons-material/FolderRounded';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { treeItemClasses } from '@mui/x-tree-view/TreeItem';
+import {
+  unstable_useTreeItem2 as useTreeItem2,
+  UseTreeItem2Parameters,
+} from '@mui/x-tree-view/useTreeItem2';
+import {
+  TreeItem2Content,
+  TreeItem2IconContainer,
+  TreeItem2Label,
+  TreeItem2Root,
+} from '@mui/x-tree-view/TreeItem2';
+import { TreeItem2Icon } from '@mui/x-tree-view/TreeItem2Icon';
+import { TreeItem2Provider } from '@mui/x-tree-view/TreeItem2Provider';
+import { TreeViewBaseItem } from '@mui/x-tree-view/models';
 
-const CustomContent = React.forwardRef(function CustomContent(
-  props: TreeItemContentProps & { lastNestedChild?: boolean },
-  ref,
-) {
-  const {
-    lastNestedChild,
-    classes,
-    className,
-    label,
-    nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
-  } = props;
+type ExtendedTreeItemProps = {
+  id: string;
+  label: string;
+  color?: 'primary' | 'default';
+};
 
-  const {
-    disabled,
-    expanded,
-    selected,
-    focused,
-    handleExpansion,
-    handleSelection,
-    preventSelection,
-  } = useTreeItem(nodeId);
+const ITEMS: TreeViewBaseItem<ExtendedTreeItemProps>[] = [
+  {
+    id: '1',
+    label: 'src',
+    color: 'primary',
+    children: [
+      {
+        id: '1.1',
+        label: 'components',
+        children: [
+          { id: '1.1.1', label: 'Button.tsx' },
+          { id: '1.1.2', label: 'Drawer.tsx' },
+          { id: '1.1.3', label: 'Navbar.tsx' },
+          { id: '1.1.4', label: 'TreeView.tsx' },
+        ],
+      },
+      {
+        id: '1.2',
+        label: 'blocks',
+        children: [
+          { id: '1.2.1', label: 'SignupPage.tsx' },
+          {
+            id: '1.2.2',
+            label: 'PricingTable',
+            children: [
+              { id: '1.2.2.1', label: 'PaymentOptions.tsx' },
+              { id: '1.2.2.2', label: 'EarlyBirdDiscount.tsx' },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+];
 
-  const icon = iconProp || expansionIcon || displayIcon;
+function DotIcon() {
+  return (
+    <Box
+      sx={{
+        width: 6,
+        height: 6,
+        borderRadius: '70%',
```

Comment:
```suggestion
        borderRadius: '50%',
```

70% feels weird 😅 Let's go with a more rounded value